### ISSUE TITLE
Verify URL incorrect with "/" restApiRoot

### ISSUE THREAD

**Issue by jonathan-chin** [strongloop/loopback#2452]
I've set my restApiRoot to "/", expecting everything to go through http://api.example.com (instead of http://www.example.com/api)

everything seems to work right except that the url generated for verification emails includes a double slash:

http://api.example.com//users/verify etc etc

It's obviously expecting not to be run at the root level. I also tried passing in an empty string "" in the options of verify but the comparison operator interpreted it as false.

how can I avoid the double slash in the generated verify url if my restapiroot is "/"?

**Comment by gunjpan** [comment]
@jonathan-chin : Hi, looks like you have stumbled upon a valid bug as there seems to be no handling of `/` as restApiRoot in creating verify link ( [code](https://github.com/strongloop/loopback/blob/45e523d66c629b3a389b84b01d28f83adbd0a1fd/common/models/user.js#L394-L405)):
```javascript
options.verifyHref = options.verifyHref ||
      options.protocol +
      '://' +
      options.host +
      displayPort +
      options.restApiRoot +
      userModel.http.path +  // for User model this will be: `/users`
      userModel.sharedClass.findMethodByName('confirm').http.path +
      '?uid=' +
      options.user.id +
      '&redirect=' +
      options.redirect;
```

This would be a simple fix. Would you like to submit a patch? I will be happy to assist to get it landed.
Meanwhile, we will be planning to get it fixed, stay tuned. Thank you.

**Comment by jonathan-chin** [comment]
I fixed it with a simple ternary comparison:

```
var restApiRoot = options.restApiRoot == "/" ? "" : options.restApiRoot;
options.verifyHref = options.verifyHref ||
options.protocol +
'://' +
options.host +
':' +
options.port +
restApiRoot +
userModel.http.path +
userModel.sharedClass.find('confirm', true).http.path +
'?uid=' +
options.user.id +
'&redirect=' +
options.redirect;
```

However, I'm still on version 2.22.2. I'd like to submit a patch for 3.0 but I'd have to set everything up for it.

**Comment by gunjpan** [comment]
@jonathan-chin : You can follow the contribution guideline and cloning/forking from GitHub would give you 3.x. Let me know, should you need further help. Thanks.

**Comment by bajtos** [comment]
Fixed & released in `loopback@2.34.1`.
